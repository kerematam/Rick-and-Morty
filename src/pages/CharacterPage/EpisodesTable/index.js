import React from 'react'
import PropTypes from 'prop-types'
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  LinearProgress,
} from '@material-ui/core'

import styles from './EpisodesTable.module.scss'

const EpisodesTable = React.memo(({ episodes, loading }) => {
  return (
    <TableContainer component={Paper}>
      {loading && <LinearProgress></LinearProgress>}
      <Table className={styles.table} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Episode </TableCell>
            <TableCell>Name</TableCell>
            <TableCell>Air Date</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {episodes.map(({ id, episode, name, air_date, url }) => (
            <TableRow hover key={id}>
              <TableCell component="th" scope="row">
                {episode}
              </TableCell>
              <TableCell>{name}</TableCell>
              <TableCell>{air_date}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
})

EpisodesTable.propTypes = {
  episodes: PropTypes.array,
  loading: PropTypes.bool,
}

export default EpisodesTable
