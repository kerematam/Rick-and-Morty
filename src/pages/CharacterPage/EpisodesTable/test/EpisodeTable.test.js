import React from 'react'
import renderer from 'react-test-renderer'
import EpisodesTable from '../index'

const setup = setupProps => {
  const wrapper = renderer.create(<EpisodesTable {...setupProps} />).toJSON()

  return {
    wrapper,
  }
}

describe('EpisodesTable', () => {
  test('should render correctly', () => {
    const mockProps = { episodes: [], loading: false }
    const { wrapper } = setup(mockProps)
    expect(wrapper).toMatchSnapshot()
  })
})
