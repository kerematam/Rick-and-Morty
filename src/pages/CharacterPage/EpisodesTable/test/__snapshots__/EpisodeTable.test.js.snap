// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesTable should render correctly 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="simple table"
    className="MuiTable-root table"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Episode 
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Air Date
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    />
  </table>
</div>
`;
